### PR TITLE
FIX: removing last component deletes entity as well

### LINKED ISSUE #82: `remove_component()` method removes entity if no more components left
**Describe the bug**
As for Single Responsibility Principle remove component should do just that but it removes empty entity as well.
As I'm creating tests for my own project, I need to populate entity with some dummy component to prevent that and test cleaning another component logic.

**To Reproduce**
```python3
import esper

class Component:
    ...


world = esper.World()
entity = world.create_entity(Component())

assert world.entity_exists(entity=entity)

world.remove_component(entity=entity, component_type=Component)

assert world.entity_exists(entity=entity)  # AssertionError
```

**Expected behavior**
I assume that cleaning components should not clear entity. As removing and adding component should be independent from entity existence. Entity should be deleted by `delete_entity()` only and not under the hood so programmer has full control over it.

**Development environment:**
 - Python 3.11
 - Esper 2.4

**Additional context**
I will create unit tests for it and fix if approved for implementation

### PR DESCRIPTION
**Pull request by gretkierewicz**
fixes #82

Due to boy scout rule, there are additional minor changes:
- add more explicit exceptions
- change return to removed component instead of entity

### CODE REVIEW

**Comment by gretkierewicz** [comment]
@benmoran56  check this out pls 

I'm using pop() instead of del to return removed component but it should not have big performance impact overall

**Review by benmoran56 — CHANGES_REQUESTED** [review]
Looks good overall, but please rollback the `if` checks in `remove_component`.

File: esper/__init__.py [hunk 1]
```diff
@@ -323,25 +324,32 @@ def add_component(self, entity: int, component_instance: _C, type_alias: _Option
...
         """Remove a Component instance from an Entity, by type.
 
-        A Component instance can be removed by providing its type.
+        A Component instance can only be removed by providing its type.
         For example: world.delete_component(enemy_a, Velocity) will remove
         the Velocity instance from the Entity enemy_a.
 
         Raises a KeyError if either the given entity or Component type does
         not exist in the database.
         """
```
[the remark this answers is not in the document]
Comment (gretkierewicz) [reply esper/__init__.py]: Sure, done

**Review by benmoran56 — APPROVED** [review]
Thank you very much!

**Comment by benmoran56** [comment]
I'll try to get a point release out soon.

**Pull request merged**